chia/wallet/vault/vault_wallet.py

```diff
@@ -0,0 +1,606 @@
+    PathHint,
+    SignedTransaction,
+    SigningInstructions,
+    SigningResponse,
+    Spend,
+    SumHint,
+)
+from chia.wallet.transaction_record import TransactionRecord
+from chia.wallet.util.compute_hints import compute_spend_hints_and_additions
+from chia.wallet.util.transaction_type import TransactionType
+from chia.wallet.util.tx_config import CoinSelectionConfig, TXConfig
+from chia.wallet.util.wallet_sync_utils import fetch_coin_spend
+from chia.wallet.vault.vault_drivers import (
+    construct_p2_delegated_secp,
+    construct_secp_message,
+    construct_vault_merkle_tree,
+    get_p2_singleton_puzzle,
+    get_p2_singleton_puzzle_hash,
+    get_recovery_finish_puzzle,
+    get_recovery_inner_puzzle,
+    get_recovery_puzzle,
+    get_recovery_solution,
+    get_vault_full_puzzle,
+    get_vault_full_solution,
+    get_vault_hidden_puzzle_with_index,
+    get_vault_inner_puzzle,
+    get_vault_inner_puzzle_hash,
+    get_vault_inner_solution,
+    get_vault_proof,
+)
+from chia.wallet.vault.vault_info import RecoveryInfo, VaultInfo
+from chia.wallet.vault.vault_root import VaultRoot
+from chia.wallet.wallet import Wallet
+from chia.wallet.wallet_info import WalletInfo
+from chia.wallet.wallet_protocol import GSTOptionalArgs
+
+
+class Vault(Wallet):
+    @staticmethod
+    async def create(
+        wallet_state_manager: Any,
+        info: WalletInfo,
+        name: str = __name__,
+    ) -> Vault:
+        self = Vault()
+        self.wallet_state_manager = wallet_state_manager
+        self.wallet_info = info
+        self.wallet_id = info.id
+        self.log = logging.getLogger(name)
+        return self
+
+    async def get_new_puzzle(self) -> Program:
+        dr = await self.wallet_state_manager.get_unused_derivation_record(self.id())
+        hidden_puzzle_hash = get_vault_hidden_puzzle_with_index(dr.index).get_tree_hash()
+        puzzle = get_vault_inner_puzzle(
+            self.vault_info.pubkey,
+            self.wallet_state_manager.constants.GENESIS_CHALLENGE,
+            hidden_puzzle_hash,
+            self.recovery_info.bls_pk if self.vault_info.is_recoverable else None,
+            self.recovery_info.timelock if self.vault_info.is_recoverable else None,
+        )
+        return puzzle
+
+    async def get_new_puzzlehash(self) -> bytes32:
+        puzzle = await self.get_new_puzzle()
+        return puzzle.get_tree_hash()
+
+    async def generate_signed_transaction(
+        self,
+        amount: uint64,
+        puzzle_hash: bytes32,
+        tx_config: TXConfig,
+        fee: uint64 = uint64(0),
+        coins: Optional[Set[Coin]] = None,
+        primaries: Optional[List[Payment]] = None,
+        memos: Optional[List[bytes]] = None,
+        puzzle_decorator_override: Optional[List[Dict[str, Any]]] = None,
+        extra_conditions: Tuple[Condition, ...] = tuple(),
+        **kwargs: Unpack[GSTOptionalArgs],
+    ) -> List[TransactionRecord]:
+        raise NotImplementedError("vault wallet")
```

Comment:
I don't know if this was intentional but we definitely want to implement this even if it just calls `generate_signed_vault_spend` under the hood.